ledger/src/blockstore.rs

```diff
@@ -6637,4 +6668,49 @@ pub mod tests {
 
         Blockstore::destroy(&blockstore_path).expect("Expected successful database destruction");
     }
+
+    #[test]
+    fn test_clear_unconfirmed_slot() {
+        let blockstore_path = get_tmp_ledger_path!();
+        {
+            let blockstore = Blockstore::open(&blockstore_path).unwrap();
+            let unconfirmed_slot = 9;
+            let unconfirmed_child_slot = 10;
+            let slots = vec![2, unconfirmed_slot, unconfirmed_child_slot];
+
+            // Insert into slot 9, mark it as dead
+            let shreds: Vec<_> = make_chaining_slot_entries(&slots, 1)
+                .into_iter()
+                .flat_map(|x| x.0)
+                .collect();
+            blockstore.insert_shreds(shreds, None, false).unwrap();
+            // Should only be one shred in slot 9
+            assert!(blockstore
+                .get_data_shred(unconfirmed_slot, 0)
+                .unwrap()
+                .is_some());
+            assert!(blockstore
+                .get_data_shred(unconfirmed_slot, 1)
+                .unwrap()
+                .is_none());
+            blockstore.set_dead_slot(unconfirmed_slot).unwrap();
+
+            // Purge the slot
+            blockstore.clear_unconfirmed_slot(unconfirmed_slot);
+            assert!(!blockstore.is_dead(unconfirmed_slot));
+            assert_eq!(
+                blockstore
+                    .meta(unconfirmed_slot)
+                    .unwrap()
+                    .unwrap()
+                    .next_slots,
+                vec![unconfirmed_child_slot]
+            );
+            assert!(blockstore
+                .get_data_shred(unconfirmed_slot, 0)
+                .unwrap()
+                .is_none());
```

Comment:
how about adding an assertion `dead_slots` are purged as well?

Reply:
@ryoqun, this: https://github.com/solana-labs/solana/pull/9805/files#diff-9ed453cbc107691c26b7e060d05e2ab0R6700 is equivalent.